### ISSUE TITLE
MAS Bridge

### ISSUE THREAD

**Issue by theonegalen** [Boris-Barboris/AtmosphereAutopilot#24]
Hi Boris, I'm looking to create an autopilot prop using https://github.com/MOARdV/AvionicsSystems/.

What I'd like to do is create a bridge where MAS can send commands to AA and display settings from AA on props. I've tried looking at your code, and I can't really make heads or tails of the different projects. Basically, I'm looking to duplicate the GUI inputs through MAS props inside the IVA: 

- Master On / Off (and tell MAS that it's on or off)
- Cruise Flight Mode: MAS sets "Level," "Course", or "Waypoint" mode, reads which mode it's in
- set waypoint, desired course, speed control, altitude hold, vertical speed control 
- Fly By Wire Mode on/off (and tell MAS that it's on or off)

Where would I find these functions in the AA code?

**Comment by Boris-Barboris** [comment]
Get AA modules for the vessel: https://github.com/Boris-Barboris/AtmosphereAutopilot/blob/master/AtmosphereAutopilot/AtmosphereAutopilot.cs#L249
If it returned null, initialize AA for the vessel: https://github.com/Boris-Barboris/AtmosphereAutopilot/blob/master/AtmosphereAutopilot/AtmosphereAutopilot.cs#L202
After that the main entry point is TopModuleManager controller wich is the root of the tree of controllers:
https://github.com/Boris-Barboris/AtmosphereAutopilot/blob/master/AtmosphereAutopilot/Modules/TopModuleManager.cs

Master On/Off is this property of topModuleManager: https://github.com/Boris-Barboris/AtmosphereAutopilot/blob/master/AtmosphereAutopilot/Modules/AutopilotModule.cs#L79

Changing autopilots from FBW to cruise flight is changing an active_controller of TopModuleManager: https://github.com/Boris-Barboris/AtmosphereAutopilot/blob/master/AtmosphereAutopilot/Modules/TopModuleManager.cs#L177

setting waypoinnt/desired I think you'll find what you want in CruiseController.cs

**Comment by Boris-Barboris** [comment]
https://github.com/RCrockford/kOS-AtmosphereAutopilot

this may be of help

**Comment by theonegalen** [comment]
Thank you

**Comment by MOARdV** [comment]
Yes, thank you.  That info will definitely help when I get started on the MAS interface.
